File: app/src/main/java/com/rolandvitezhu/todocloud/network/helper/RetrofitResponseHelper.java
package com.rolandvitezhu.todocloud.network.helper;

import com.google.gson.GsonBuilder;
import com.rolandvitezhu.todocloud.network.api.dto.BaseResponse;

import retrofit2.Response;

public class RetrofitResponseHelper {

  public static String ResponseToJson(Response response) {
    try {
      return new GsonBuilder().disableHtmlEscaping().serializeNulls().create().toJson(response.body());
    } catch (Exception e) {
      return e.getMessage();
    }
  }

  public static boolean IsNoError(Response response) {
    if (response.body() != null) {
      BaseResponse baseResponse = (BaseResponse) response.body();

      return baseResponse.getError().equals("false");
    } else {
      return false;
    }
  }
}

File: app/src/main/java/com/rolandvitezhu/todocloud/network/helper/RetrofitResponseHelper.kt
package com.rolandvitezhu.todocloud.network.helper

import com.google.gson.GsonBuilder
import com.rolandvitezhu.todocloud.network.api.dto.BaseResponse
import retrofit2.Response

open class RetrofitResponseHelper {

    companion object {
        fun ResponseToJson(response: Response<*>): String? {
            return try {
                GsonBuilder().disableHtmlEscaping().serializeNulls().create().toJson(response.body())
            } catch (e: Exception) {
                e.message
            }
        }

        fun IsNoError(response: Response<*>): Boolean {
            return if (response.body() != null) {
                val baseResponse = response.body() as BaseResponse?
                baseResponse!!.getError() == "false"
            } else {
                false
            }
        }
    }
}